battleship/battleship-single.py

```diff
@@ -0,0 +1,368 @@
+
+            # Get start coordinate and check validity
+            start = input('Coordinate for the start of your ' +
+                          shipNames[i] + '? ')
+            startCoords = convert(start)
+            if startCoords is None:
+                print(
+                    'Invalid input. Examples of proper coordinates: a0, B9, c2. Please try again.')
+                continue
+
+            # Get end coordinate and check validity
+            end = input('Coordinate for the end of your ' +
+                        shipNames[i] + '? ')
+            endCoords = convert(end)
+            if endCoords is None:
+                print(
+                    'Invalid input. Examples of proper coordinates: a0, B9, c2. Please try again.')
+                continue
+
+            # Check valid ship length and ensure they are all on one row/column
+            if startCoords[0] == endCoords[0]:
+                if abs(startCoords[1]-endCoords[1]) + 1 != shipLength[i]:
+                    print(
+                        'Invalid ship length. Please try again.')
+                    continue
+            elif startCoords[1] == endCoords[1]:
+                if abs(startCoords[0]-endCoords[0]) + 1 != shipLength[i]:
+                    print(
+                        'Invalid ship length. Please try again.')
+                    continue
+            else:
+                print(
+                    'The start and end of the ship must be on the same row or column. Please try again.')
+                continue
+
+            # Check validity of ship placement
+            if checkShipPlacement(startCoords, endCoords, i, playerShipAliveCoords, playerGrid):
+                printGrid(playerGrid, True)
+                break
+            else:
+                print('Ships cannot intersect. Please try again.')
+                continue
+
+    print("Ships successfully set!")
+    sleep(2)
+    print("--------------------------------------------------")
+    print("START GAME\n")
+    sleep(2)
+
+
+def setComputerShips(shipNames, shipLength, computerShipAliveCoords):
+    '''Set the ship placements of the computer'''
+
+    # Loops through ships
+    for i in range(len(shipNames)):
+        while True:
+            # Get random start coordinate
+            startCoords = (randint(0, 9), randint(0, 9))
+            # Get random end coordinate from start coordinate to ensure proper ship length
+            endCoords = tuple(map(add, startCoords, choice(
+                [(0, shipLength[i]-1), (shipLength[i]-1, 0)])))
+            # Ensure end coordinate is on the grid
+            if endCoords[0] < 0 or endCoords[0] > 9 or endCoords[1] < 0 or endCoords[1] > 9:
+                continue
+            # Check validity of ship placement
+            if checkShipPlacement(startCoords, endCoords, i, computerShipAliveCoords):
+                print(str(startCoords) + str(endCoords))
```

Comment:
This debugging line reveals the position of all of the computer’s ships, so probably don’t leave it in lol

Reply:
Oh CRAP I forgot to take that out LOL